### PR TITLE
Planemo type annotation: config, context and factory

### PR DESCRIPTION
**Pull request by gallardoalba**

### CODE REVIEW

File: planemo/config.py [hunk 1]
```diff
@@ -119,14 +131,14 @@ def callback(ctx, param, value):
     return option
 
 
-def global_config_path(config_path=None):
+def global_config_path(config_path: Optional[str] = None) -> str:
     if not config_path:
         config_path = os.environ.get(PLANEMO_CONFIG_ENV_PROP, "~/.planemo.yml")
         config_path = os.path.expanduser(config_path)
```
Comment (nsoranzo) [inline planemo/config.py:137]:
```suggestion
        planemo_config_env_prop = os.environ.get(PLANEMO_CONFIG_ENV_PROP, "~/.planemo.yml")
        config_path = os.path.expanduser(planemo_config_env_prop)
```

File: planemo/config.py [hunk 2]
```diff
@@ -4,11 +4,21 @@
 from enum import Enum
 from typing import (
     Any,
+    Callable,
     Dict,
+    List,
+    Optional,
+    Tuple,
+    Union,
+    TYPE_CHECKING,
```
Comment (nsoranzo) [inline planemo/config.py:13]:
```suggestion
    TYPE_CHECKING,
    Union,
```

**Comment by gallardoalba** [comment]
Thanks!

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Thanks @gallardoalba !

**Comment by gallardoalba** [comment]
> Thanks @gallardoalba !

Thanks for the review Nicola, finally you contributed more than me to this PR.

**Pull request merged**